File: src/main/java/org/mifos/fineract/services/CurrencyApi.java
package org.mifos.fineract.services;

import org.mifos.fineract.models.GetCurrenciesResponse;
import org.mifos.fineract.models.PutCurrenciesRequest;
import org.mifos.fineract.models.PutCurrenciesResponse;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.PUT;
import rx.Observable;

public interface CurrencyApi {
    /**
     * Retrieve Currency Configuration
     * Returns the list of currencies permitted for use AND the list of currencies not selected (but available for selection).  Example Requests:  currencies   currencies?fields&#x3D;selectedCurrencyOptions
     *
     * @return Observable&lt;GetCurrenciesResponse&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("currencies")
    Observable<GetCurrenciesResponse> retrieveCurrencies();


    /**
     * Update Currency Configuration
     * Updates the list of currencies permitted for use.
     *
     * @param body body (required)
     * @return Observable&lt;PutCurrenciesResponse&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @PUT("currencies")
    Observable<PutCurrenciesResponse> updateCurrencies(
            @retrofit2.http.Body PutCurrenciesRequest body
    );

}

File: src/main/kotlin/org/openapitools/client/apis/CurrencyApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.GetCurrenciesResponse
import org.openapitools.client.models.PutCurrenciesRequest
import org.openapitools.client.models.PutCurrenciesResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PUT

interface CurrencyApi {
    /**
     * Retrieve Currency Configuration
     * Returns the list of currencies permitted for use AND the list of currencies not selected (but available for selection).  Example Requests:  currencies   currencies?fields&#x3D;selectedCurrencyOptions
     * Responses:
     *  - 200: OK
     *
     * @return [GetCurrenciesResponse]
     */
    @GET("v1/currencies")
    suspend fun retrieveCurrencies(): GetCurrenciesResponse

    /**
     * Update Currency Configuration
     * Updates the list of currencies permitted for use.
     * Responses:
     *  - 200: OK
     *
     * @param putCurrenciesRequest
     * @return [PutCurrenciesResponse]
     */
    @PUT("v1/currencies")
    suspend fun updateCurrencies(@Body putCurrenciesRequest: PutCurrenciesRequest): PutCurrenciesResponse

}
